<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1178</width>
    <height>360</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="widget" native="true">
      <widget class="QLineEdit" name="lineEdit">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>641</width>
         <height>61</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
      </widget>
      <widget class="QWidget" name="horizontalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>641</width>
         <height>61</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout"/>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>70</y>
         <width>82</width>
         <height>161</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="pushButton_one">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_four">
          <property name="text">
           <string>4</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_seven">
          <property name="text">
           <string>7</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_x">
          <property name="text">
           <string>x</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="verticalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>70</y>
         <width>82</width>
         <height>161</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButton_two">
          <property name="text">
           <string>2</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_five">
          <property name="text">
           <string>5</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_eight">
          <property name="text">
           <string>8</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_zero">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="verticalLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>70</y>
         <width>82</width>
         <height>161</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QPushButton" name="pushButton_three">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_six">
          <property name="text">
           <string>6</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_nine">
          <property name="text">
           <string>9</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_dot">
          <property name="text">
           <string>.</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>70</y>
         <width>371</width>
         <height>161</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_division">
          <property name="text">
           <string>/</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pushButton_mod">
          <property name="text">
           <string>Mod</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_plus">
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="pushButton_openBracket">
          <property name="text">
           <string>(</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="pushButton_clear">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="pushButton_closeBracket">
          <property name="text">
           <string>)</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_multiply">
          <property name="text">
           <string>*</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="pushButton_backspace">
          <property name="text">
           <string>Backspace</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButton_minus">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QPushButton" name="pushButton_equals">
          <property name="text">
           <string>=</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="pushButton_power">
          <property name="text">
           <string>^</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>230</y>
         <width>143</width>
         <height>41</height>
        </rect>
       </property>
       <property name="title">
        <string>    Enter a value for X</string>
       </property>
       <widget class="QLineEdit" name="variable_x">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>20</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QPushButton" name="pushButton_sin">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>240</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Sin</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_cos">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>240</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Cos</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_tan">
       <property name="geometry">
        <rect>
         <x>370</x>
         <y>240</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Tan</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_asin">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>280</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Asin</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_acos">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>280</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Acos</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_atan">
       <property name="geometry">
        <rect>
         <x>370</x>
         <y>280</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Atan</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_sqrt">
       <property name="geometry">
        <rect>
         <x>490</x>
         <y>240</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Sqrt</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_log">
       <property name="geometry">
        <rect>
         <x>490</x>
         <y>270</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Log</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_ln">
       <property name="geometry">
        <rect>
         <x>490</x>
         <y>300</y>
         <width>100</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Ln</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_graph">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>290</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton#pushButton_graph {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #0099cc, stop: 1 #007acc); /* Lighter gradient background */
    color: white; /* Text color */
    border: 1px solid #0099cc; /* Border style */
    border-radius: 15px; /* Increased border radius for a more rounded button */
    padding: 5px 15px; /* Padding around the text */
}

QPushButton#pushButton_graph:hover {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #007acc, stop: 1 #0099cc); /* Gradient background on hover */
    border: 1px solid #007acc; /* Border style on hover */
    color: #f0f0f0; /* Text color on hover */
}
</string>
       </property>
       <property name="text">
        <string>Graph</string>
       </property>
      </widget>
      <widget class="QCustomPlot" name="qcustomplot" native="true">
       <property name="geometry">
        <rect>
         <x>659</x>
         <y>-1</y>
         <width>491</width>
         <height>281</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_plot">
       <property name="geometry">
        <rect>
         <x>1010</x>
         <y>290</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton#pushButton_plot {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #0099cc, stop: 1 #007acc); /* Gradient background */
    color: white; /* Text color */
    border: 1px solid #0099cc; /* Border style */
    border-radius: 10px; /* Slightly rounded corners */
    padding: 5px 15px; /* Padding around the text */
}

QPushButton#pushButton_plot:hover {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #007acc, stop: 1 #0099cc); /* Gradient background on hover */
    border: 1px solid #007acc; /* Border style on hover */
    color: #f0f0f0; /* Text color on hover */
}
</string>
       </property>
       <property name="text">
        <string>Plot</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_xmax">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>290</y>
         <width>58</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>x max =</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_xmin">
       <property name="geometry">
        <rect>
         <x>670</x>
         <y>290</y>
         <width>58</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>x min =</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_ymax">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>310</y>
         <width>58</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>y max =</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_ymin">
       <property name="geometry">
        <rect>
         <x>670</x>
         <y>310</y>
         <width>58</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>y min =</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="x_min">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>290</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-1000000</number>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>-10</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="y_min">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>310</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-10000000</number>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>-10</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="x_max">
       <property name="geometry">
        <rect>
         <x>890</x>
         <y>290</y>
         <width>81</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-1000000</number>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="y_max">
       <property name="geometry">
        <rect>
         <x>890</x>
         <y>310</y>
         <width>81</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-1000000</number>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_clear_graph">
       <property name="geometry">
        <rect>
         <x>1010</x>
         <y>321</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton#pushButton_clear_graph{
    border: 2px solid #8f8f91;
    border-radius: 10px;
    background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #f6f7fa, stop:1 #dadbde);
    color: black;
}
QPushButton:pressed#pushButton_clear_graph{
    background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #dadbde, stop:1 #f6f7fa);
}</string>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
